tests/core/test_lightning_optimizer.py

```diff
@@ -195,6 +195,7 @@ def configure_optimizers(self):
     assert sgd_zero_grad.call_count == 10
 
 
+@pytest.mark.filterwarnings(r"ignore:Detected call of `lr_scheduler.step\(\)` before `optimizer.step\(\)`:UserWarning")
```

Comment:
Since the optimizers are mocked in this test case, the actual `.step()` has never been called, and thus, it raises the warning. so, I'll just ignore the warning here.

Reply:
can you add a brief comment there in the test about the reason for skipping? :)